app/models/product_import/entry_validator.rb

```diff
@@ -283,7 +287,11 @@ def product_field_errors(entry, existing_product)
     end
 
     def attributes_match?(attribute, existing_product, entry)
-      existing_product.public_send(attribute) == entry.public_send(attribute)
+      if existing_product.public_send(attribute).is_a?(Numeric) || entry.public_send(attribute).is_a?(Numeric)
+        return existing_product.public_send(attribute).to_i == entry.public_send(attribute).to_i
+      else
+        return existing_product.public_send(attribute).to_s == entry.public_send(attribute).to_s
+      end
```

Comment:
dear ruby experts, there must be a better way to do this, I just want to compare ignoring the type...

Reply:
Why not compare only with `to_s` ? If one is numeric it'll be transformed into a string anyway.

Reply:
Yes, it was my first try but some numbers will be converted to "1.0" and others to "1" and to_s compare will fail.

Reply:
Oh dear, it looks like we are allowing too much flexibility. The existing product should always have the same type for an attribute, right? What about the CSV? Always string? If the specs supply other types, it's the problem of the spec.

I think you are generalising too much here. `Numeric` can be float like `2.1` but you just convert to integer, loosing the decimals: `2`.

If you have one trusted value (from the database) and one untrusted value (import) then I would always convert the untrusted value into the format of the trusted value and compare on that basis. So if we read an *on demand* column, we should convert that into a boolean and then compare. If we read a stock level, then as integer. Price as float and so on. If that fails we need to rescue and notify the user.

Reply:
ok, I have a new improved version that handles floats and only converts the entry to the attribute type, not the other way around.
what do you think @mkllnk ?